servicetalk-tcp-netty-internal/src/main/java/io/servicetalk/tcp/netty/internal/TcpServerConfig.java

```diff
@@ -24,24 +24,41 @@
 
 import java.net.SocketOption;
 import java.net.StandardSocketOptions;
+import java.time.Duration;
 import java.util.HashMap;
 import java.util.Map;
 import javax.annotation.Nullable;
 
 import static io.servicetalk.transport.netty.internal.SocketOptionUtils.addOption;
+import static io.servicetalk.utils.internal.DurationUtils.ensureNonNegative;
+import static java.time.Duration.ofSeconds;
 import static java.util.Objects.requireNonNull;
 
 /**
  * Configuration for TCP based servers.
  */
 public final class TcpServerConfig extends AbstractTcpConfig<ServerSslConfig> {
+    /**
+     * The maximum length of client hello message as defined by
+     * <a href="https://www.rfc-editor.org/rfc/rfc5246#section-6.2.1">RFC5246</a>.
+     */
+    private static final int MAX_CLIENT_HELLO_LENGTH = 0xFFFFFF;
```

Comment:
@normanmaurer Can you point me where `0xFFFFFF` value comes from?
I followed the RFC link and it says `2^14` instead of `2^24 - 1`.

Source: https://github.com/netty/netty/commit/535da17e45201ae4278c0479e6162bb4127d4c32#diff-1c7c1e64a0afabaa56b28c9fd8a3c9173a8904b2cee64be327563b7d97b69086R41-R45

Reply:
It's been a while so I dont remember exactly but its also mentioned in other RFCs , for example: https://datatracker.ietf.org/doc/html/rfc6347#section-3.2.3.

Reply:
also that said I am not sure if this is a good default ... we were just very "conservative" as we didn't want to break people (there was no limit before). I think in practice using something much smaller is better.

Reply:
Thanks!
After reading https://www.rfc-editor.org/rfc/rfc5246#section-7.4.1.2, it doesn't look like ClientHello specifically has a limit but I'm agreed that TLS in general defines a limit for its messages as 2^24 - 1.

Agreed that it's not a perfect default. At least, it's better than unlimited. We will start with this for backward compatibility and we will try to reduce it after we experiment in apps.